### ISSUE TITLE
master store minion in redis or other backends

### ISSUE THREAD

**Issue by derek0377** [saltstack/salt#34238]
Hi,we have 10000 minions, but master store minions key in local file system,read and write speed is limited, Do have some other backends like redis?

Thank you!

**Comment by Ch3LL** [comment]
@derek0377 this is currently not available to specify a different backend for the keys. This is related to #25284  and #5752 which the requests are similar. But I will keep this one open to track this for adding the ability to manage keys in redis. Thanks

**Comment by derek0377** [comment]
@Ch3LL , sorry to disturb, I think support multi-backbends with session and token function for salt-api is good! As really do not use single salt-master,would perfer standby or cluster, Do salt support multi-backends for session and token of salt-api? 
thanks!

**Comment by whiteinge** [comment]
@derek0377 not at the moment. Pluggable backend storage for sharing eauth keys (in addition to minion keys in OP) would be a great addition to Salt.

If you have an immediate need and if you don't mind hacking your own solution see the conversation below for some ideas:

https://gitter.im/cherrypy/cherrypy?at=576962e9c417e36c69e4c697

**Comment by cachedout** [comment]
This is actually on our short list for features. Hopefully we'll have some movement here soon.

**Comment by whiteinge** [comment]
@cachedout is there a single issue to track that addition? IMO it would be helpful to enumerate all the needed things. The pki dir is an obvious addition, but there's more than one cache (grains, mine, files, etc), and eauth tokens will be an easy one to overlook/forget.

**Comment by OrangeDog** [comment]
@cachedout this should probably be reopened

**Comment by cachedout** [comment]
@OrangeDog I agree but I don't have any GitHub permissions to do so. :) @sagetherage ?

**Comment by Akm0d** [comment]
Reopening based on **https://github.com/saltstack/salt/issues/57449**

**Comment by Deshke** [comment]
cool, is there any progress on this ?
